micrometer-observation/src/main/java/io/micrometer/observation/Observation.java

```diff
@@ -933,15 +933,15 @@ class Context implements ContextView {
          * @return name
          */
         @Override
-        public String getName() {
+        public @Nullable String getName() {
             return this.name;
         }
 
         /**
          * Sets the observation name.
          * @param name observation name
          */
-        public void setName(String name) {
+        public void setName(@Nullable String name) {
```

Comment:
Is it valid? I mean if you set the name should you set it as `null`?
(I guess somewhere we use this and pass `null`?)

But now I think it's symmetric with `setContextualName`.

Reply:
I think you're right it shouldn't be `@Nullable`. We need to change `SimpleObservation`'s package-private constructor arg for name to be not nullable for that, which is fine since it is package private and all call sites don't pass nullable values. I'll update the PR.